gridded_etl_tools/utils/zarr_methods.py

```diff
@@ -901,8 +915,24 @@ def calculate_update_time_ranges(
 
         return datetime_ranges, regions_indices
 
-    def is_append_contiguous(self, original_dataset: xr.Dataset, append_times: list) -> bool:
-        """Checks that an append will produce a contiguous dataset
+    def are_times_contiguous(self, times: list[datetime.datetime], expected_delta: datetime.timedelta) -> bool:
+        """
+        Check if the given list of times is contiguous with a given difference between each time.
+
+        @param times            A list of datetime objects
+        @param expected_delta   Amount of time expected to be between each time
+        @return                 True if times are contiguous and have the correct time step, False otherwise
+        """
+        previous_time = times[0]
+        for time in times[1:]:
+            if time - previous_time != expected_delta:
+                return False
+            previous_time = time
```

Comment:
Assigns previous_time the current time so that next iteration the next time will check against this one. The loop in general is iterating over times in order and checking if the previous time was one time step behind it.